drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -567,20 +550,6 @@ static int nrf5_tx(const struct device *dev,
 
 	LOG_DBG("Result: %d", nrf5_data.tx_result);
 
-#if NRF_802154_ENCRYPTION_ENABLED
-	/*
-	 * When frame encryption by the radio driver is enabled, the frame stored in
-	 * the tx_psdu buffer is:
-	 * 1) authenticated and encrypted in place which causes that after an unsuccessful
-	 *    TX attempt, this frame must be propagated back to the upper layer for retransmission.
-	 *    The upper layer must ensure that the exact same secured frame is used for
-	 *    retransmission
-	 * 2) frame counters are updated in place and for keeping the link frame counter up to date,
-	 *    this information must be propagated back to the upper layer
-	 */
-	memcpy(payload, nrf5_radio->tx_psdu + 1, payload_len);
```

Comment:
@e-rk I'm afraid we can't remove this `memcpy` because of the second reason.

Reply:
That's not good. This will still break the frames, because it might be encrypted multiple times.

Reply:
I have removed this change for the time being.

Reply:
Thanks. We are assuming invalid retransmitted frames anyway, but at least operation without retransmissions won't be compromised.